Stop converting the filename to lowercase

sigmavirus24 opened the pull request:
This leads to problems for tools integrating with pydocstyle, e.g., atom
linter-pydocstyle

Closes #179 

cc @BrnoPCmaniak

Nurdok commented:
Notice that the tests fail for the PR - see the Travis build.
Also, this should be checked on a Windows machine - I think the `lower` call was made because there where "c:\" - "C:\" incosistencies. I'll have to check this later.

sigmavirus24 commented:
> Notice that the tests fail for the PR - see the Travis build.

I did notice that and I was attempting to determine why before I had to leave yesterday. :)

> Also, this should be checked on a Windows machine 

Should we attempt to set up AppVeyor for pydocstyle?

For whatever it may be worth, pycodestyle doesn't munge the paths for case sensitivity and doesn't appear to have problems on Windows.

sigmavirus24 commented:
So I updated the test case that was failing. I also updated the tests for a few things I noticed (that I'll happily pull into a separate pull request) including parametrizing the test that was failing and fixing a warning that pytest wasn't printing by default (and printing pytest warings by default).

Nurdok commented on src/tests/test_integration.py:
```diff
@@ -21,7 +21,7 @@
 
 
-class TestEnv(object):
+class FakeTestEnv(object):
```
Why this name change?

sigmavirus24 replied:
So I'm happy to pull d99705e into a separate pull request. I just noticed it and fixed it while working on this.

Nurdok replied:
Oh, this is because `pytest` tries to collect this class? Then it's fine to change the name. However, I don't like "fake" because is creates a real file system environment. How about `InvocationEnv` / `StagingEnv` / `DeployEnv`, etc.?

The-Compiler replied:
Yup - pytest tries to collect classes named `Test*`

Nurdok commented:
I merged this manually and changed `FakeTestEnv` to `SandboxEnv`. Thanks!

sigmavirus24 commented:
Sorry about that. I was going to get to it tonight. :)
